app/notebook/server/CalcWebSocketService.scala

```diff
@@ -43,6 +45,9 @@ class CalcWebSocketService(
     var calculator: ActorRef = null
     var wss: List[WebSockWrapper] = Nil
 
+    protected val notebookStartTime = LocalTime.now()
+    private var lastCellExecutionTime: Option[LocalTime] = None
```

Comment:
the only thing I'm fearing here is that things are working in an observable or a model is currently being trained, so that 60 minutes might be too little.
Or if you go to lunch :)